src/agent/solana/exporter.rs

```diff
@@ -657,17 +671,65 @@ impl Exporter {
             compute_unit_price_micro_lamports = Some(price);
         }
 
-        // If the dynamic unit price is enabled, use the estimated price if it is higher
-        // than the curdynamic_compute_unit_pricing_enabled
-        if let Some(estimated_recent_price) = self.recent_compute_unit_price_micro_lamports {
-            // Get the estimated compute unit price and Wrap it so it stays below the maximum total
-            // compute unit fee
-            let estimated_recent_price = estimated_recent_price
-                .min(MAXIMUM_TOTAL_COMPUTE_UNIT_FEE_MICRO_LAMPORTS / total_compute_limit as u64);
+        if self.config.dynamic_compute_unit_pricing_enabled {
+            let maximum_unit_price =
+                MAXIMUM_TOTAL_COMPUTE_UNIT_FEE_MICRO_LAMPORTS / total_compute_limit as u64;
+
+            // Use the estimated previous price if it is higher
+            // than the current price.
+            if let Some(estimated_recent_price) = self.recent_compute_unit_price_micro_lamports {
+                // Get the estimated compute unit price and wrap it so it stays below the maximum total
+                // compute unit fee. We additionally divide such price by 2 to make sure a spike
+                // doesn't get propagated forever.
+                let estimated_price = (estimated_recent_price >> 1).min(maximum_unit_price);
+
+                compute_unit_price_micro_lamports = compute_unit_price_micro_lamports
+                    .map(|price| price.max(estimated_price))
+                    .or(Some(estimated_price));
+            }
+
+            // Use exponentially higher price if this publisher hasn't published in a while for the accounts
+            // in this batch. This will use the maximum total compute unit fee if the publisher
+            // hasn't updated for >= MAXIMUM_SLOT_GAP_FOR_DYNAMIC_COMPUTE_UNIT_PRICE slots.
```

Comment:
this is pretty much guaranteed to use the maximum price when the market opens each day right?

I think that behavior is fine -- I think the max price isn't so high that we can't afford 1 of these transactions per day per price feed or whatever -- though probably worth a comment justifying it.

Reply:
Yeah it was a certain challenge. I think we should see how it will work with the other dynamic mechanism (which is decaying) and make sure it won't get very expensive.